topics/go/algorithms/strings/reverse/reverse.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import "fmt"
+
+func main() {
+	// Run ReverseString
+	fmt.Println(ReverseString("Hello World"))
+}
+
+func ReverseString(str string) string {
+	// Create rune.
+	chars := []rune(str)
+
+	// Reverse String.
+	for start, end := 0, len(chars)-1; start < end; start, end = start+1, end-1 {
```

Comment:
This is a place where a comment would be helpful. Describe while the complicated look and the semantics of how this works. This is a place where you should be teaching.

Remember, you are working with code points (which are not characters). UTF-8 is tricky.

Reply:
I rewrote this to use a simpler form that is easier to understand.